src/TabsPage.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+import { Tab } from './types/Tab';
+
+interface TabsPageProps {
+  tabs: Tab[];
+}
+
+export const TabsPage: React.FC<TabsPageProps> = ({ tabs }) => {
+  const { tabId } = useParams();
+
+  const findTabItem = (id: string) => {
+    return tabs.find(tab => tab.id === id);
+  };
+
+  const checkTabContent = (id: string) => {
+    const tabItem = findTabItem(id);
+    const tabContent = tabItem ? tabItem.content : 'No content';
+
+    return tabContent;
+  };
+
+  const setActiveClassTab = (id: string) => {
+    return classNames({
+      'is-active': tabId === id,
+    });
+  };
```

Comment:
The creation of the `findTabItem`, `checkTabContent` and `setActiveClassTab` functions can be moved outside the component and used as helpers. 
This change will ensure that these functions will not be created every time the component is rerendered.